### PR TITLE
[qtbase] enable egl for android

### PR DESCRIPTION
**Pull request by m-kuhn**
Enable support for feature `egl` for `android` for `qtbase`.
Specifying it as default-feature for android but at the same time specifying the feature as not supported is inconsistent.

This actively expects `qtbase:*-android` to fail in CI, so far they were ignored because of the above-mentioned inconsistency.

Branched off https://github.com/microsoft/vcpkg/pull/35845 which is more complicated because it requests changes to the ci infrastructure.

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Please handle the build error on CI pipeline.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -933,9 +933,15 @@ qt5-canvas3d:x64-windows=skip
...
 qt5-canvas3d:x86-windows=skip
+# Missing Android SDK in Docker image, see https://github.com/microsoft/vcpkg/pull/35845
+qtbase:arm64-android=fail
+qtbase:arm-neon-android=fail
+qtbase:x64-android=fail
 # Missing system libraries
 qtwayland:x64-osx=fail
 qtwayland:arm64-osx=fail
+# Missing system libraries when pulled in as dependency for android
+qtwayland:x64-linux=skip
```
Comment (m-kuhn) [inline scripts/ci.baseline.txt:944]: Should be fixed by https://github.com/microsoft/vcpkg/pull/35961/files#diff-783e7bf9703fa77a2652b514ee607d8e40aafdd4f25f7e4e663de8b224f43e62R36 but I cannot prove it in this PR, as the docker image in use is a prebuilt one

**Comment by m-kuhn** [comment]
@LilyWangLL this needs a change in the base docker image for the android ci pipeline. How can this be tested?

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (m-kuhn) [reply scripts/ci.baseline.txt:944]:
```suggestion
```

**Comment by m-kuhn** [comment]
Also needs changes to the ci, abandoning this one to have only one discussion at https://github.com/microsoft/vcpkg/pull/35845.

**Pull request closed without merging**